tools/ARIAtools/extractProduct.py

```diff
@@ -267,20 +267,48 @@ def merged_productbbox(product_dict, workdir='./', bbox_file=None, croptounion=F
     if not os.path.exists(workdir):
         os.mkdir(workdir)
 
+    # Load bbox_file if given
+    if bbox_file:
+        user_bbox=open_shapefile(bbox_file, 0, 0)
+
     # Extract/merge productBoundingBox layers
+    validation_array=[]
     for scene in product_dict:
         # Get pair name, expected in dictionary
         pair_name=scene["pair_name"][0]
         outname=os.path.join(workdir, pair_name+'.shp')
 
         # Create union of productBoundingBox layers
+        frames_in_scene=0 # keep running tally of number of valid frames
         for frame in scene["productBoundingBox"]:
             prods_bbox=open_shapefile(frame, 'productBoundingBox', 1)
+
+            # Check if frames fall within bounds of bbox file
+            if bbox_file:
+                # find area of intersection between user bbox and frame extent
+                prod_in_user_bbox=prods_bbox.intersection(user_bbox)
+
+                # report percent of frame within user bbox
+                per_in_user_bbox=prod_in_user_bbox.area/prods_bbox.area*100
+
+                # remove frame if area less than some limit
+                if per_in_user_bbox>min_percent_frame_overlap:
+                    frames_in_scene+=1
+
```

Comment:
@rzinke From a quick read through the code, i would not do the percentage overlap check in individual products. Rather would do the check on the union of the products for a given InSAR pair. @sssangha to provide some additional discussion.